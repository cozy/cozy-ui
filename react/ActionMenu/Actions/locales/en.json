{
  "viewInContacts": "View in Cozy Contacts",
  "modify": "Modify",
  "emailTo": "Send an email",
  "smsTo": "Send a message",
  "call": "Call"
}
