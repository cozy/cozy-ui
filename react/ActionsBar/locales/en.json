{
  "selected": "%{smart_count} item selected |||| %{smart_count} items selected"
}
