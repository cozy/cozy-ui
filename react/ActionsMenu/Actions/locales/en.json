{
  "viewInContacts": "View in Cozy Contacts",
  "viewInDrive": "Open",
  "modify": "Modify",
  "emailTo": "Send an email",
  "smsTo": "Send a message",
  "print": "Print",
  "others": "Others",
  "call": "Call"
}
