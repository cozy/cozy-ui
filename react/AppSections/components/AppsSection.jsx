import React from 'react'
import { translate } from '../../I18n'
import withBreakpoints from '../../helpers/withBreakpoints'
import { getTranslatedManifestProperty } from '../helpers'
import sortBy from 'lodash/sortBy'
import SmallAppItem from './SmallAppItem'
import styles from './AppsSection.styl'

const makeNameGetter = t => app => getTranslatedManifestProperty(app, 'name', t)

export const AppsSection = ({
  t,
  appsList,
  subtitle,
  onAppClick,
  IconComponent,
  breakpoints = {}
}) => {
  const { isMobile } = breakpoints
  return (
    <div className={styles.AppsSection}>
      {subtitle}
      {appsList && !!appsList.length && (
        <div className={styles.AppsSection__list}>
          {sortBy(appsList, makeNameGetter(t)).map(app => (
            <SmallAppItem
              app={app}
              namePrefix={getTranslatedManifestProperty(app, 'name_prefix', t)}
              name={getTranslatedManifestProperty(app, 'name', t)}
              onClick={() => onAppClick(app.slug)}
              key={app.slug}
              isMobile={isMobile}
              IconComponent={IconComponent}
            />
          ))}
        </div>
      )}
    </div>
  )
}

export default translate()(withBreakpoints()(AppsSection))
