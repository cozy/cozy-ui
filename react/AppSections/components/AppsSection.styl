.AppsSection 
    margin-bottom 1rem

.AppsSection__list
    display flex
    flex-wrap wrap
    justify-content flex-start
