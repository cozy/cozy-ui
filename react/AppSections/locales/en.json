{
  "app_categories": {
    "all": "All categories",
    "banking": "Banking",
    "cozy": "The essentials",
    "education": "Education",
    "energy": "Energy",
    "health": "Health",
    "host_provider": "Host",
    "insurance": "Insurance",
    "isp": "ISP",
    "konnectors": "Services",
    "list_separator": ", ",
    "mes_infos": "MesInfos experimentation",
    "online_services": "Online services",
    "others": "Others",
    "partners": "Partners",
    "press": "Press",
    "productivity": "Productivity",
    "ptnb": "PTNB experimentation",
    "public_service": "Public services",
    "shopping": "Shopping",
    "social": "Social",
    "telecom": "Telecom",
    "transport": "Transportation"
  },
  "sections": {
    "applications": "Applications",
    "konnectors": "Services"
  }
}
