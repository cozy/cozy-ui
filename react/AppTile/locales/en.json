{
  "app_item": {
    "by": "By",
    "installed": "Installed",
    "maintenance": "In maintenance",
    "update": "Update available",
    "favorite": "Added to home page"
  }
}
