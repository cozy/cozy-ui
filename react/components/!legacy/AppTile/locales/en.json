{
  "app_item": {
    "by": "By",
    "installed": "Installed",
    "maintenance": "In maintenance",
    "update": "Update available"
  }
}
