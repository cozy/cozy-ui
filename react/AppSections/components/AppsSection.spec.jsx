'use strict'

/* eslint-env jest */

import React from 'react'
import { mount } from 'enzyme'

import { CozyProvider } from 'cozy-client'

import mockApps from '../../mocks/apps'
import en from '../locales/en'
import { I18nContext } from '../../jestLib/I18n'
import Tile, { TileTitle, TileSubtitle, TileFooter } from '../../Tile'
import { BreakpointsProvider } from '../../hooks/useBreakpoints'
import { I18n } from '../../I18n'

import { AppsSection } from './AppsSection'

const i18nContext = I18nContext({ locale: en })
const tMock = i18nContext.t

describe('AppsSection component', () => {
  let component
  const client = {}
  const setup = ({ onAppClick }) => {
    component = mount(
      <CozyProvider client={client}>
        <BreakpointsProvider>
          <I18n lang="en" dictRequire={() => en}>
            <AppsSection
              t={tMock}
              lang="en"
              subtitle={<h3>Test Apps</h3>}
              appsList={mockApps}
              onAppClick={onAppClick}
            />
          </I18n>
        </BreakpointsProvider>
      </CozyProvider>
    ).find(AppsSection)
  }

  it('should be rendered correctly with apps list, subtitle', () => {
    const mockOnAppClick = jest.fn()
    setup({ onAppClick: mockOnAppClick })
    expect(
      component.find(Tile).map(x => {
        const developer = x.find(TileSubtitle)
        const status = x.find(TileFooter)
        return {
          title: x.find(TileTitle).text(),
          developer: developer.length ? developer.text() : null,
          status: status.length ? status.text() : null
        }
      })
    ).toMatchSnapshot()
  })

  it('should run provided onAppClick on AppTile click event', () => {
    const mockOnAppClick = jest.fn()
    setup({ onAppClick: mockOnAppClick })
    expect(component.find(Tile).length).toBe(mockApps.length)
    const appItem = component.find(Tile).at(0)
    appItem.simulate('click')
    expect(mockOnAppClick.mock.calls.length).toBe(1)
    expect(mockOnAppClick.mock.calls[0][0]).toBe('konnector-bouilligue')
  })
})
