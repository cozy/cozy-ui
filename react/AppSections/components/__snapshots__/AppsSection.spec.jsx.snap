// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsSection component should be rendered correctly with apps list, subtitle 1`] = `
Array [
  Object {
    "developer": "By Cozy",
    "status": null,
    "title": "Bouilligue",
  },
  Object {
    "developer": "By Cozy",
    "status": "Installed",
    "title": "Cozy Collect",
  },
  Object {
    "developer": "By Cozy",
    "status": null,
    "title": "DevOnly",
  },
  Object {
    "developer": "By Cozy",
    "status": "Installed",
    "title": "Cozy Drive",
  },
  Object {
    "developer": "By Cozy",
    "status": null,
    "title": "Cozy Photos",
  },
  Object {
    "developer": null,
    "status": "Update available",
    "title": "Tasky",
  },
  Object {
    "developer": "By Cozy",
    "status": "Installed",
    "title": "Trinlane",
  },
]
`;
